### ISSUE TITLE
Remove outdated options from neighbourhood plugins

### ISSUE THREAD

**Issue by cgsandford** [metoppv/improver#482]
As an IMPROVER developer I want a clear neighbourhood processing interface so that it is easy to work with.

After #480 there will be some arguments to neighbourhood plugins that are no longer required (including ens_factor and re_mask).  These arguments should be removed from the plugin code.

Related issues: #480 (prerequisite), #352 

Acceptance criteria:
 * Identify and remove arguments to the neighbourhood processing plugins that are no longer required
 * Update unit tests

**Comment by fionaRust** [comment]
Superseded by metoppv/mo-blue-team#157
